.github/workflows/ci.yml

```diff
@@ -15,10 +15,28 @@ on:
 
 jobs:
 
-  CI:
-    uses: ./.github/workflows/merge-gate.yml
-    secrets: inherit
+  checks:
+    name: Checks
+    uses: ./.github/workflows/checks.yml
 
-  Metrics:
-    needs: [ CI ]
+  tests:
+    name: Tests
+    uses: ./.github/workflows/tests_with_coverage.yml
+
+  metrics:
+    needs: [ tests, checks ]
     uses: ./.github/workflows/report.yml
+
+  # This job ensures inputs have been executed successfully.
+  approve-merge:
+    name: Allow Merge
+    runs-on: ubuntu-latest
+    # If you need additional jobs to be part of the merge gate, add them below
+    needs: [ metrics ]
```

Comment:
Allow merge should not depend on report / metrics. 
We still have incidents in which metrics / KPIs / report fails for some reason and want to be able to develop and merge PRs, though. 
So for now: creating the reports is optional and should not prevent to merge.

```suggestion
    needs: [ checks, tests ]
```

To make dependencies clear, I propose to move job `metrics` to the end of the file.